File: packages/sdk/dotnet/Modules/MemoryApi.cs
using System.Text.Json;

namespace Ihui.AI;

/// <summary>
/// 记忆模块 — 保存 / 召回 / 搜索 / Dream / 遗忘 / 分类记忆。
/// </summary>
/// <remarks>
/// 端点(8 个):
/// <list type="bullet">
///   <item>POST   /v1/memory(保存记忆)</item>
///   <item>GET    /v1/memory(召回记忆)</item>
///   <item>POST   /v1/memory/search(语义搜索)</item>
///   <item>POST   /v1/memory/dream(Dream 梦境系统)</item>
///   <item>DELETE /v1/memory(遗忘记忆)</item>
///   <item>GET    /v1/memory/working(工作记忆)</item>
///   <item>GET    /v1/memory/episodic(情景记忆)</item>
///   <item>GET    /v1/memory/procedural(程序记忆)</item>
/// </list>
/// </remarks>
public sealed class MemoryApi
{
    private readonly BaseClient _client;

    internal MemoryApi(BaseClient client) => _client = client;

    /// <summary>POST /v1/memory(保存记忆)。</summary>
    /// <param name="req">请求体(content / type / metadata 等)</param>
    /// <param name="cancellationToken">取消令牌</param>
    /// <returns>保存结果(JsonElement,含 memoryId / status)</returns>
    public Task<JsonElement?> SaveAsync(object req, CancellationToken cancellationToken = default)
        => _client.RequestAsync<JsonElement>("POST", "/memory", req, cancellationToken);

    /// <summary>GET /v1/memory(召回记忆)。</summary>
    /// <param name="cancellationToken">取消令牌</param>
    /// <returns>记忆列表(JsonElement)</returns>
    public Task<JsonElement?> RecallAsync(CancellationToken cancellationToken = default)
        => _client.RequestAsync<JsonElement>("GET", "/memory", null, cancellationToken);

    /// <summary>POST /v1/memory/search(语义搜索)。</summary>
    /// <param name="req">搜索请求</param>
    /// <param name="cancellationToken">取消令牌</param>
    /// <returns>搜索结果(JsonElement)</returns>
    public Task<JsonElement?> SearchAsync(MemorySearchRequest req, CancellationToken cancellationToken = default)
        => _client.RequestAsync<JsonElement>("POST", "/memory/search", req, cancellationToken);

    /// <summary>POST /v1/memory/dream(Dream 梦境系统)。</summary>
    /// <param name="req">请求体(可为空对象)</param>
    /// <param name="cancellationToken">取消令牌</param>
    /// <returns>Dream 结果(JsonElement)</returns>
    public Task<JsonElement?> DreamAsync(object req, CancellationToken cancellationToken = default)
        => _client.RequestAsync<JsonElement>("POST", "/memory/dream", req, cancellationToken);

    /// <summary>DELETE /v1/memory(遗忘记忆)。</summary>
    /// <param name="req">请求体(memoryId)</param>
    /// <param name="cancellationToken">取消令牌</param>
    /// <returns>遗忘结果(JsonElement,含 memoryId / status)</returns>
    public Task<JsonElement?> ForgetAsync(object req, CancellationToken cancellationToken = default)
        => _client.RequestAsync<JsonElement>("DELETE", "/memory", req, cancellationToken);

    /// <summary>GET /v1/memory/working(工作记忆)。</summary>
    /// <param name="cancellationToken">取消令牌</param>
    /// <returns>工作记忆(JsonElement)</returns>
    public Task<JsonElement?> WorkingAsync(CancellationToken cancellationToken = default)
        => _client.RequestAsync<JsonElement>("GET", "/memory/working", null, cancellationToken);

    /// <summary>GET /v1/memory/episodic(情景记忆)。</summary>
    /// <param name="cancellationToken">取消令牌</param>
    /// <returns>情景记忆(JsonElement)</returns>
    public Task<JsonElement?> EpisodicAsync(CancellationToken cancellationToken = default)
        => _client.RequestAsync<JsonElement>("GET", "/memory/episodic", null, cancellationToken);

    /// <summary>GET /v1/memory/procedural(程序记忆)。</summary>
    /// <param name="cancellationToken">取消令牌</param>
    /// <returns>程序记忆(JsonElement)</returns>
    public Task<JsonElement?> ProceduralAsync(CancellationToken cancellationToken = default)
        => _client.RequestAsync<JsonElement>("GET", "/memory/procedural", null, cancellationToken);
}

File: packages/sdk/java/src/main/java/com/ihui/ai/sdk/module/MemoryApi.java
package com.ihui.ai.sdk.module;

import com.fasterxml.jackson.databind.JsonNode;
import com.ihui.ai.sdk.BaseClient;
import com.ihui.ai.sdk.model.MemorySearchRequest;

import java.util.Map;

/**
 * 记忆模块 — 保存 / 召回 / 搜索 / Dream / 遗忘 / 分类记忆。
 *
 * <p>端点(8 个):
 * <ul>
 *   <li>POST   /v1/memory(保存记忆)</li>
 *   <li>GET    /v1/memory(召回记忆)</li>
 *   <li>POST   /v1/memory/search(语义搜索)</li>
 *   <li>POST   /v1/memory/dream(Dream 梦境系统)</li>
 *   <li>DELETE /v1/memory(遗忘记忆)</li>
 *   <li>GET    /v1/memory/working(工作记忆)</li>
 *   <li>GET    /v1/memory/episodic(情景记忆)</li>
 *   <li>GET    /v1/memory/procedural(程序记忆)</li>
 * </ul>
 */
public final class MemoryApi {

    private final BaseClient client;

    /**
     * 构造 MemoryApi。
     *
     * @param client 底层 BaseClient
     */
    public MemoryApi(BaseClient client) {
        this.client = client;
    }

    /**
     * POST /v1/memory(保存记忆)。
     *
     * @param req 请求体(content / type / metadata 等)
     * @return 保存结果(JsonNode,含 memoryId / status)
     */
    public JsonNode save(Map<String, Object> req) {
        return client.request("POST", "/memory", req, JsonNode.class);
    }

    /**
     * GET /v1/memory(召回记忆)。
     *
     * @return 记忆列表(JsonNode)
     */
    public JsonNode recall() {
        return client.request("GET", "/memory", null, JsonNode.class);
    }

    /**
     * POST /v1/memory/search(语义搜索)。
     *
     * @param req 搜索请求
     * @return 搜索结果(JsonNode)
     */
    public JsonNode search(MemorySearchRequest req) {
        return client.request("POST", "/memory/search", req, JsonNode.class);
    }

    /**
     * POST /v1/memory/dream(Dream 梦境系统)。
     *
     * @param req 请求体(可为空 Map)
     * @return Dream 结果(JsonNode)
     */
    public JsonNode dream(Map<String, Object> req) {
        return client.request("POST", "/memory/dream", req, JsonNode.class);
    }

    /**
     * DELETE /v1/memory(遗忘记忆)。
     *
     * @param req 请求体(memoryId)
     * @return 遗忘结果(JsonNode,含 memoryId / status)
     */
    public JsonNode forget(Map<String, Object> req) {
        return client.request("DELETE", "/memory", req, JsonNode.class);
    }

    /**
     * GET /v1/memory/working(工作记忆)。
     *
     * @return 工作记忆(JsonNode)
     */
    public JsonNode working() {
        return client.request("GET", "/memory/working", null, JsonNode.class);
    }

    /**
     * GET /v1/memory/episodic(情景记忆)。
     *
     * @return 情景记忆(JsonNode)
     */
    public JsonNode episodic() {
        return client.request("GET", "/memory/episodic", null, JsonNode.class);
    }

    /**
     * GET /v1/memory/procedural(程序记忆)。
     *
     * @return 程序记忆(JsonNode)
     */
    public JsonNode procedural() {
        return client.request("GET", "/memory/procedural", null, JsonNode.class);
    }
}
